From c3582855b88cfde8e69734da738803b54c2c2e26 Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Tue, 17 Mar 2020 07:25:12 +0100
Subject: [PATCH] Fix bug #79329 - get_headers should not accept \0

From 0d139c5b94a5f485a66901919e51faddb0371c43

(cherry picked from commit b7b9302660a23a67285e204bc3d7fcf6ba7f6533)
(cherry picked from commit b9a1e6bfd762d2bf7fa3c5bbcfbb6dcdfdfa982c)
---
 ext/standard/url.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/standard/url.c b/ext/standard/url.c
index 6ecace53e5..d6e71fa487 100644
--- a/ext/standard/url.c
+++ b/ext/standard/url.c
@@ -675,7 +675,7 @@ PHP_FUNCTION(get_headers)
 	HashTable *hashT;
 	long format = 0;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "s|l", &url, &url_len, &format) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "p|l", &url, &url_len, &format) == FAILURE) {
 		return;
 	}
 	context = FG(default_context) ? FG(default_context) : (FG(default_context) = php_stream_context_alloc(TSRMLS_C));
